### PR TITLE
Upload page design

### PR DESCRIPTION
**Pull request by dansterenson**

### CODE REVIEW

**Review by BenSterenson — APPROVED** [review]

File: client/src/components/sideBar.vue [hunk 1]
```diff
@@ -0,0 +1,63 @@
+<template >
+  <v-navigation-drawer
+        v-model="drawer"
+        :mini-variant.sync="mini"
+        permanent
+        dark
+        color="1A1A1A"
```
Comment (MPalarya) [inline client/src/components/sideBar.vue:7]: you can use the newly introduced variable

File: client/src/components/uploadExcel.vue [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+export default {
+
+  data: function() {
+    return {
+      myFiles: [],
+      myServer: {
+        async process(fieldName, file) {
+          const formData = new FormData();
+          formData.append("file", file);
+          const idToken = await firebase.auth.currentUser.getIdToken(/* forceRefresh */ true);
```
Comment (MPalarya) [inline client/src/components/uploadExcel.vue:51]: can we reuse getAuthToken from mainContent?

File: client/src/components/viewDashboard.vue [hunk 3]
```diff
@@ -0,0 +1,152 @@
+<template>
```
Comment (MPalarya) [inline client/src/components/viewDashboard.vue:1]: is this a draft file? it looks like a copy-paste of an old version of mainContent

File: client/src/components/mainContent.vue [hunk 4]
```diff
@@ -134,6 +134,9 @@ export default {
     async onPageLoad() {
       const idToken = await this.getAuthToken();
       await this.getPortfolioData(idToken);
+      if(!this.DataWasLoaded){
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply client/src/components/mainContent.vue:137]: Added Eslint to the package.json file. so now when you compile the files, It executes Eslint

File: client/src/components/navBar.vue [hunk 5]
```diff
@@ -1,6 +1,6 @@
 <template>
-  <v-app-bar app color="#1A1A1A" dark>
-    <v-toolbar-title>Financex</v-toolbar-title>
+  <v-app-bar app dark clipped-left>
+    <v-toolbar-title class="font-weight-light" style="font-size: 250%">Vestika</v-toolbar-title>
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply client/src/components/navBar.vue:3]: Fixed

**Review by MPalarya — APPROVED** [review]

File: client/src/app.vue [hunk 6]
```diff
@@ -1,9 +1,12 @@
...
   <v-app id="app">
     <nav-bar v-if="!isRegistrationRoute()" />
-    <side-bar v-if="!isRegistrationRoute()" />
+    <side-bar
+      v-if="!isRegistrationRoute()"
+      :is-another-upload="isAnotherUpload"
+    />
     <v-main id="main">
-      <router-view />
+      <router-view @another="check" />
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply client/src/app.vue:9]: Ill fix it

File: client/src/components/uploadExcel.vue [hunk 7]
```diff
@@ -29,7 +29,7 @@
         max-file-size="10MB"
         style-item-panel-aspect-ratio="0.3"
         :allow-multiple="false"
-        :server="myServer"
+        :server="{process}"
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply client/src/components/uploadExcel.vue:32]: This is the syntax, It allows me to add more functions to the Filepond server, according to this: [https://github.com/pqina/vue-filepond/issues/15](url)

File: client/src/views/mainContent.vue [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  beforeMount() {
+    this.fetchData(1)
+  },
+  methods: {
+    async sleep(ms) {
+      return new Promise((resolve) => setTimeout(resolve, ms));
+    },
+    async fetchData(numRetries) {
+        const idToken = await FireGetToken();
+        if (localStorage.portfoliosData) {
```
[the remark this answers is not in the document]
Comment (dansterenson) [reply client/src/views/mainContent.vue:48]: No big difference, I can change it
They just behave differently when no key/value pair is found with the requested name. For example, if key 'item' does not exist, var a = localStorage.item; will result in a being undefined, while var a = localStorage.getItem('item'); will result in a having the value null. 
[https://stackoverflow.com/questions/12632463/is-localstorage-getitemitem-better-than-localstorage-item-or-localstoragei](url)

**Pull request merged**